.github/workflows/ci-e2e.yml

```diff
@@ -3,6 +3,12 @@ name: ci-e2e
 on:
   pull_request:
     branches: [master]
+  workflow_dispatch:
+
+# Required for allowing PRs to master branch to be of highest priority
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: true
```

Comment:
Very minor nit: could the indentation be adjusted for easier reading here?

Reply:
Also just for clarification, how do we indicate that this is the same workflow as `trigger-e2e`?

Reply:
Hmm I think the indentation cannot be adjusted further as the `group` and `cancel-in-progress` keys are siblings. This is similar to the [example from the docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-concurrency-to-cancel-any-in-progress-job-or-run).

Also, this ci-e2e workflow will run differently from the trigger-e2e workflow. To summarise the points:
1. Multiple trigger-e2e workflows can run at the same time, because it functions somewhat like a queue system where they will regularly poll to see if there are any pending or in progress runs of the ci-e2e workflow. It can happen if multiple people were to trigger the e2e tests from different pull requests.
2. If a ci-e2e workflow is currently running, the few trigger-e2e workflows that are running will keep waiting and polling until they find a window where there are no ci-e2e workflow running.
3. If a ci-e2e workflow is currently running and we were to perform a release (defined as creating a PR to the master branch), then any existing ci-e2e runs will be terminated to make way for the e2e tests to be performed on the release.

There is a very edge case where 2 trigger-e2e workflows manage to trigger a ci-e2e workflow run. In that case, GitHub Actions will take the latest one and the earlier one will be terminated. If that happens, the user would not be notified, but the terminated e2e test can run again by sending the command to trigger e2e tests again.